Mathlib/RingTheory/Polynomial/HilbertPoly.lean

```diff
@@ -58,9 +58,29 @@ equals `(n - k + d).choose d`.
 noncomputable def preHilbertPoly (d k : ℕ) : F[X] :=
   (d.factorial : F)⁻¹ • ((ascPochhammer F d).comp (Polynomial.X - (C (k : F)) + 1))
 
+lemma natDegree_preHilbertPoly [CharZero F] (d k : ℕ) :
+    (preHilbertPoly F d k).natDegree = d := by
+  have hne : (d ! : F) ≠ 0 := by norm_cast; positivity
+  rw [preHilbertPoly, natDegree_smul _ (inv_ne_zero hne), natDegree_comp, ascPochhammer_natDegree,
+    add_comm_sub, ← C_1, ← map_sub, natDegree_add_C, natDegree_X, mul_one]
+
+lemma coeff_preHilbertPoly_self [CharZero F] (d k : ℕ) :
+    (preHilbertPoly F d k).coeff d = (d ! : F)⁻¹ := by
```

Comment:
Is this then the leadingCoeff? Please add a lemma with that phrasing, unless you want to replace this one